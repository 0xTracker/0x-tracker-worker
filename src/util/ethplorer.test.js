const config = require('config');

const { configure, getTokenInfo } = require('./ethplorer');

beforeAll(() => {
  configure(config.get('ethplorer'));
});

describe('ethplorer.getTokenInfo', () => {
  it('should fetch info for recognised token', async () => {
    const tokenInfo = await getTokenInfo(
      '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
    );

    expect(tokenInfo).toMatchSnapshot();
  });

  it('should return null for unrecognised token', async () => {
    const tokenInfo = await getTokenInfo('0x9200983');

    expect(tokenInfo).toBeNull();
  });

  it('should return null for invalid address', async () => {
    const tokenInfo = await getTokenInfo('harbard');

    expect(tokenInfo).toBeNull();
  });
});
