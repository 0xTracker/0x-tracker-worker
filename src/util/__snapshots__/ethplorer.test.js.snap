// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ethplorer.getTokenInfo should fetch info for recognised token 1`] = `
Object {
  "address": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
  "decimals": 18,
  "name": "Wrapped Ether",
  "symbol": "WETH",
}
`;
